javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -20,15 +26,43 @@ public final class OpenTelemetryInstaller {
    * @return the {@link AutoConfiguredOpenTelemetrySdk}
    */
   public static AutoConfiguredOpenTelemetrySdk installOpenTelemetrySdk(
-      ClassLoader extensionClassLoader) {
+      ClassLoader extensionClassLoader, EarlyInitAgentConfig earlyConfig) {
 
     AutoConfiguredOpenTelemetrySdk autoConfiguredSdk =
         AutoConfiguredOpenTelemetrySdk.builder()
             .setResultAsGlobal()
             .setServiceClassLoader(extensionClassLoader)
             .build();
+    ConfigProvider configProvider = AutoConfigureUtil.getConfigProvider(autoConfiguredSdk);
     OpenTelemetrySdk sdk = autoConfiguredSdk.getOpenTelemetrySdk();
 
+    setForceFlush(sdk);
+
+    if (configProvider != null) {
+      return SdkAutoconfigureAccess.create(
```

Comment:
If I understand correctly this ends up creating a new `AutoConfiguredOpenTelemetrySdk`. There should be a comment spelling out that this is being done and an explanation why it is done. If it turns out that using `Resource.getDefault()` here is ok then that should also be explained. To me this seems fishy because in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/3e1647c9397cc88bd296d87e4b3064fc1e935aab/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java#L178 we'd read the attributes from the default resource which doesn't seem correct to me.